@()(implicit session: Session)
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <title>MetaQ home</title>
    <link href = "https://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel = "stylesheet">
    <link rel='stylesheet' type='text/css' href="@routes.Assets.at("stylesheets/home.css")">
    <script src="@routes.Assets.at("javascripts/jquery-2.2.4.min.js")" type="text/javascript"></script>
    <script src="@routes.Assets.at("javascripts/bootstrap.min.js")" type="text/javascript"></script>
    <script src='@routes.Assets.at("javascripts/homePage.js")' type='text/javascript'></script>
  </head>
  <body>
    <div class="user">
    @if(session.get("username").isDefined) {
      <h5>Hi, @session.get("username") . </h5>
      <h5 class="log-link"><a href="@routes.Application.logout()">Log out <i class="glyphicon glyphicon-log-out"></i> </a></h5>
    } else {
      <p>You are not logged in</p>
      <h5 class="log-link"><a href="@routes.Application.login()">Log in <i class="glyphicon glyphicon-log-in"></i></a></h5>
      <h5 class="signup"><a href="@routes.Application.newUser()">Sign up </a></h5>
    }
    </div>
    <div class="title">
      <h1>MetaQ - Explore metagenomes</h1>
    </div>

    <div class="choose-button">
      <ul class="nav nav-tabs " role="tablist"  id="tabs">
        <li role="presentation" class="active"><a data-target="#genePanel" aria-controls="genePanel" role="tab" data-toggle="tab">
          <h4>Explore ORF Products </h4></a>
        </li>
        <li role="presentation"><a data-target="#pwayPanel" aria-controls="pwayPanel" role="tab" data-toggle="tab">
          <h4>Explore Pathways  </h4></a>
        </li>
      </ul>
      <div class="tab-content well">

        <div role="tabpanel" class="tab-pane active" id="genePanel">
          <div class="nav-bar">
            <div class="search">
              <form class="navbar-form" id="searchGene" data-submit="@routes.Application.geneExplorer("")">
                <div class="input-group add-on">
                  <input class="form-control" id="searchGeneBox" placeholder="Search protein products" type="text">
                  <button class="btn btn-default" type="submit"><i class="glyphicon glyphicon-search"></i></button>
                </div>
              </form>
              <div style="height:20px;width:70px">
                <a id="settings-toggle">Advanced search...</a>
              </div>
            </div>

            <div class="settings" style="display:none">
              <div>
                <p>Search by:</p>
              </div>
              <div class="radio-inline">
                <label><input type="radio" name="df" value="product" checked="checked">Protein product (default)</label>
              </div>
              @*<div class="radio-inline">*@
                @*<label><input type="radio" name="df" value="extended_desc">Secondary Description</label>*@
              @*</div>*@
              <div class="radio-inline">
                <label><input type="radio" name="df" value="ORFID">ORF id</label>
              </div>
              <div class="radio-inline">
                <label><input type="radio" name="df" value="pway">Pathway ID name</label>
              </div>
              <div class="checkbox-inline">
                <label><input type="checkbox" name="hq" value="true" >Return only high quality hits</label>
              </div>
              <div style="padding-right:0">
                <p>Only return hits above </p>
              </div>
              <div class="input-group input-group-sm" style="width:100px">
                <input style="width:60px" type="text" id="rpkm" value="0"  aria-describedby="sizing-addon3">
                <span class="input-group-addon" id="sizing-addon3"> RPKM </span>
              </div>
            </div>
          </div>
        </div>

        <div role="tabpanel" class="tab-pane" id="pwayPanel">
          <div class="nav-bar">
            <div class="search">
              <form class="navbar-form" id="searchPway" data-submit="@routes.Application.pwayExplorer("")">
                <div class="input-group add-on">
                  <input class="form-control" id="searchPwayBox" placeholder="Search pathways" type="text">
                  <button class="btn btn-default" type="submit"><i class="glyphicon glyphicon-search"></i></button>
                </div>
              </form>
            </div>
          </div>
        </div>

      </div>
    </div>
    <div class="demo-info">
      <h3>What is MetaQ?</h3>
      <p>MetaQ is a web application tailored for navigation and exploration of <a href="http://hallam.microbiology.ubc.ca/MetaPathways/" target="_blank">MetaPathways</a> results.
        MetaQ couples a back-end database of indexed data products generated from MetaPathways with an interactive front end web service with which to explore, query,
        and visualize the indexed data. Users can make individual or shared accounts as well as export results as tab separated value (TSV) files</p>

      <h3>How does this demo work?</h3>

      <p>For this demo, We tested MetaQ using the <a href="http://www.gutcyc.org/" target="_blank">GutCyc</a> compendium which boasts 37 million ORFs, and over 1000 metabolic pathways across 400+ metagenomes.
        Each ORF is annotated against <a href="http://www.metacyc.org/" target="_blank">MetaCyc</a>, <a href="http://www.genome.jp/kegg/" target="_blank">KEGG</a>
        and <a href="http://www.ncbi.nlm.nih.gov/COG/" target="_blank">COG</a> protein databases.
      <p>MetaQ is still a work in progress, and it currently supports 2 search modules.
      The ORF Exploration module works for searches based on open reading frames, so one can search functionally annotated genes through protein product names (as default),
      as well as add filters like a minimum RPKM allowed or for results to require a functionally annotation id (from COG or KEGG) to refine the searches.
      In addition, the GeneExplorer module also includes a clustering visualization which groups up results based on common keywords between them for explorative purposes.
      On the other hand, the PathwayExplorer module works around protein pathways. A user can search what identified pathways were annotated from the data (based on their pathway name),
      and extract the ORFs associated with it, as well as what metagenome sample it was found on.</p>
      <p>Current functionality also supports exporting the results found and downloading them as a TSV file with each column attribute for further use.</p>

      <h3>What is MetaQ built on?</h3>

      <p>The backend is an Apache Solr server instance, it indexes both ORFs and pathways and serves all the data to the frontend application.
        This frontend application was built using the Play Framework, which parses user input and communicates with the backend directly.
        Complimentary data visualization in the frontend is done through JavaScript's D3 library.</p>

      @*<h3>Demo Example</h3>*@
      @*<p></p>*@
    </div>

    <div class="author">
      <h4>Created by: Jorge Luis Lazo </h4>
      <h4>Contact: <a href="jorge.lazo@@alumni.ubc.ca" target="_top">jorge.lazo@@alumni.ubc.ca </a></h4>
      <h4>Source code: <a href="https://github.com/Jlazo29/pathwayQuery" target="_blank">GitHub </a></h4>
    </div>

  </body>
</html>